app/forms/support_user_invite_form.rb

```diff
@@ -0,0 +1,51 @@
+class SupportUserInviteForm < ApplicationForm
+  FORM_PARAMS = %i[first_name last_name email].freeze
+
+  attr_accessor :email, :first_name, :last_name, :service
+
+  validate :validate_support_user
+  validates :service, presence: true
+
+  def invite!
+    save_support_user
+  end
+
+  def as_form_params
+    { "support_user" => slice(FORM_PARAMS) }
+  end
+
+  private
+
+  def support_user
+    @support_user ||=
+      begin
+        user = user_klass
+          .with_discarded
+          .discarded
+          .find_or_initialize_by(email:)
+        user.assign_attributes(first_name:, last_name:)
+        user
+      end
+  end
+
+  def save_support_user
+    ActiveRecord::Base.transaction do
+      support_user.undiscard! if support_user.discarded?
+      support_user.save!
+    end
+    SupportUser::Invite.call(support_user:)
+  end
+
+  def validate_support_user
+    if support_user.invalid?
+      support_user.errors.each do |err|
+        errors.add(err.attribute, err.message)
+      end
+    end
+  end
```

Comment:
Hmm, I wonder if we could add something like https://github.com/rails/rails/blob/36c1591bcb5e0ee3084759c7f42a706fe5bb7ca7/activerecord/lib/active_record/validations/associated.rb#L54?

Not a blocker.

Reply:
The current implementation seems more explicit, so I'm going to leave this for now.